frontend-ssr/components/Scroll/Scroll.tsx

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable @typescript-eslint/no-empty-interface */
+import React from 'react';
+import BScroll from 'better-scroll';
+import throttle from 'lodash/throttle';
+
+import styles from './Scroll.module.scss';
+
+const INFINITE_SCROLL_DISTANCE = 100;
+
+interface State {}
+
+interface Props {
+  hasInfiniteScroll: boolean;
+  /** we use the property isReversed here so we can listen to the bottom or the
+   * top of the page */
+  isReversed?: boolean;
+  /** fetch data when we reach the top of the page, maybe we will need to do that in
+   *  some components */
+  onStartReach?: () => Promise<void>;
+  /** fetch data when we reach the end of the page */
+  onEndReach: () => Promise<void>;
+  preloadDistance: number;
+  className?: string;
+}
+class Scroll extends React.Component<Props, State> {
+  forceStop: Record<string, unknown>;
+
+  scrollRef: React.RefObject<HTMLDivElement>;
+
+  bScroll: any;
+
+  constructor(props) {
+    super(props);
+    this.forceStop = {};
+    this.scrollRef = React.createRef();
+    this._checkStartReach = throttle(this._checkStartReach.bind(this), 50);
```

Comment:
We can avoid `bind` here. When I wrote this code in DevDrive I didn't know that using 

`_checkStartReach = (...) => { ... }` will preserve the `this`. So using this method - called `class fields` if I'm not mistaken - makes things more readable.

Reply:
Yes, I have recently read about this method. :D Class fields it is.